src/cpg_flow/stage.py

```diff
@@ -76,11 +76,17 @@ def __init__(
         stage: Optional['Stage'] = None,
     ):
         # Converting str into Path objects.
```

Comment:
Leaving a comment here so I stop getting GitHub notifications. Tracking this back to prod-pipes, there was a comment [here](https://github.com/populationgenomics/production-pipelines/blob/main/cpg_workflows/workflow.py#L124), `Converting str into Path objects`, which was not actually done in-code. There's a specific point on this own repository's README [here](https://github.com/populationgenomics/cpg-flow?tab=readme-ov-file#string-outputs-from-expected_outputs) stating that String and Path returns are treated differently, which is no longer true.

This has been actioned in a recent cpg-flow PR and is a behavioral change I don't think we wanted. This current PR doubles down on it, but I think reversion is preferable.

Reply:
No worries, I was pretty surprised when I went looking for the cpg_workflows state and saw that comment...